prereq-build: add extra check for elfutils required header

Ansuel opened the pull request:
While testing tools build on an alpine image it was found that with musl
libc some header are missing for elfutils tool.

Add extra prereq-build check to make sure these header are present in
the system to correctly compile host tools.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
include/prereq.mk: add RequireCHeader helper

Add RequireCHeader helper that will try to compile a fake c program with
the requested header included.

This is useful to check if a specific header is present in the system
without checking for the specific path.

This is a generilized version of the current ncurses test.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
prereq-build: add extra check for elfutils required header

While testing tools build on an alpine image it was found that with musl
libc some header are missing for elfutils tool.

Add extra prereq-build check to make sure these header are present in
the system to correctly compile host tools.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

dhewg commented on include/prereq.mk:
```diff
@@ -63,6 +63,14 @@ define RequireHeader
 
+define RequireCHeader
+  define Require/$(1)
+    printf '#include <$(1)>\n int main(){}' | gcc -x c -
```
Looks like this may write a.out to `pwd`?
For the record: prereq-build.mk uses this cmdline: `gcc -include ncurses.h -x c -o $(TMP_DIR)/a.out`

Ansuel replied:
Oh thanks! Think a better solution would be generalize the ncurses check

Ansuel replied:
And yes i notice the a.out but i tought it was part of me testing command.

Ansuel committed:
prereq-build: use RequireCHeader test for ncurses prereq

RequireCHeader is a generilized version of the ncurses prereq test. Use
that indetad.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@dhewg can you give this an extra check?

dhewg commented:
Sure, LGTM!
